### PR TITLE
Add non-atomic interval for timing report

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem

`AtomicInterval` in `sdk::timing` is convenient for timing report for both muti-thread and single thread execution. But for single thread execution, it incurs an unnecessary overhead for accessing and updating atomic variables. 

#### Summary of Changes
Add a non-atomic interval for timing single thread execution. 

Fixes #

### CODE REVIEW

File: sdk/src/timing.rs [hunk 1]
```diff
@@ -103,12 +104,54 @@ impl AtomicInterval {
...
+    }
+
+    /// return ms until the interval_time will have elapsed
+    pub fn remaining_until_next_interval(&self, interval_time: u64) -> u64 {
+        interval_time.saturating_sub(self.elapsed_ms())
+    }
+}
+
+/// Thread safe. Can be used to time single/mutli thread execution. If you are timing single thread
+/// exectuion, consider using NonAtomicInterval, the no thread safe but with better performance.
```
Comment (jeffwashington) [inline sdk/src/timing.rs:143]: "the no thread..." is not clear.

File: sdk/src/timing.rs [hunk 2]
```diff
@@ -66,11 +66,12 @@ pub fn slot_duration_from_slots_per_year(slots_per_year: f64) -> Duration {
 }
 
 #[derive(Debug, Default)]
-pub struct AtomicInterval {
-    last_update: AtomicU64,
+pub struct Interval<T> {
```
Comment (jeffwashington) [inline sdk/src/timing.rs:69]: why generic? timestamp will always return u64, right?

**Comment by jeffwashington** [comment]
Did you measure overhead?

File: sdk/src/timing.rs (hunk 2 above)
Comment (HaoranYi) [reply sdk/src/timing.rs:69]: > Did you measure overhead?

```
running 2 tests
test bench_atomic_interval     ... bench:       4,796 ns/iter (+/- 573)
test bench_non_atomic_interval ... bench:       4,801 ns/iter (+/- 440)
```
non atomic is actually slower.

Comment (HaoranYi) [reply sdk/src/timing.rs:69]: > why generic? timestamp will always return u64, right?

Yes. T is either u64 or AtomicU64 for single thread/multi-thread.

**Pull request closed without merging**